src/System.Threading.Overlapped/tests/HandleFactory.cs

```diff
@@ -35,12 +47,47 @@ private static unsafe Win32Handle CreateHandle(bool async, string fileName = nul
         p.dwSecurityQosFlags = (uint)0;
         p.lpSecurityAttributes = IntPtr.Zero;
         p.hTemplateFile = IntPtr.Zero;
-        return DllImport.CreateFile2(
-            fileName ?? @"Overlapped.tmp",
+        handle = DllImport.CreateFile2(
+            fileName,
             DllImport.FileAccess.GenericWrite,
             DllImport.FileShare.Write,
             DllImport.CreationDisposition.CreateAlways,
             &p);
 #endif
+
+        if (!handle.IsInvalid)
+        {
+            return handle;
+        }
+
+        int errorCode = Marshal.GetLastWin32Error();
+        string filePath = Path.GetFullPath(fileName);
+        string message =
+            $"CreateFile or CreateFile2 failed (error code {errorCode}): {new Win32Exception(errorCode).Message}{Environment.NewLine}" +
+            $"    File name: {fileName}{Environment.NewLine}" +
+            $"    File path: {filePath}{Environment.NewLine}";
+        if (Directory.Exists(Path.GetDirectoryName(filePath)))
+        {
+            bool success = false;
+            try
+            {
+                File.WriteAllText(filePath, string.Empty);
+                success = true;
```

Comment:
Do you actually need this bool?  Why not just append "Successfully wrote to the file" here?